// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Copyright 2017 Google, Inc.
// Author: mjansche@google.com (Martin Jansche)
//
// \file
// Serializable voice data.

syntax = "proto3";

package sweet;

message WorldFrame {
  float start = 1;
  float end = 2;

  float lf0 = 3;

  repeated double sp = 4 [packed = true];
  repeated float mgc = 5 [packed = true];

  repeated float bap = 6 [packed = true];
}

message WorldData {
  double frame_shift_s = 1;

  uint32 num_samples = 2;
  uint32 sample_rate_hz = 3;
  repeated WorldFrame frame = 4;
}
